File: LeetCode/0034_Find_First_and_Last_Position_of_Element_in_Sorted_Array/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    let first=-1;
    let second =-1;
    let i=0;
    let j=nums.length-1;
    let mid=-1;
    while(i<=j){
        mid=Math.floor((i+j)/2);
        if(nums[mid]==target) break;
        else if(target>nums[mid]) i=mid+1;
        else j=mid-1;
    }
    if(nums[mid]==target){
        first=mid;
        second=mid;
    }
    while(nums[first-1]==target) first--;
    while(nums[second+1]==target) second++;

    return [first,second]
};

File: LeetCode/0034_Find_First_and_Last_Position_of_Element_in_Sorted_Array/solution.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        if(nums.size()<1) return {-1,-1};
        int first = -1;
        int second = -1;
        int i = 0;
        int j = nums.size() - 1;
        int mid = -1;
        while (i <= j) {
            mid = ((i + j) / 2);
            if (nums[mid] == target)
                break;
            else if (target > nums[mid])
                i = mid + 1;
            else
                j = mid - 1;
        }
        if (nums[mid] == target) {
            first = mid;
            second = mid;
        }
        while (first>0 && nums[first - 1] == target)
            first--;
        while (second<nums.size()-1 && nums[second + 1] == target)
            second++;

        vector <int> st = {first,second};

        return st;
    }
};